### PR TITLE
Upgrade RESTEasy to 3.7.0.Final

### PR DESCRIPTION
**Pull request by ronsigal**
[WFLY-11987] Update feature-pack modules.

[WFLY-11987] resteasy-validator-provider-11 -> resteasy-validator-provider

in pom.xml, feature-pack/pom.xml

[WFLY-11987] Fix circular dependency wrt resteasy-jackson2-provider in galleon-pack

[WFLY-11987] Fix circular dependency wrt resteasy-jackson2-provider

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by ronsigal** [comment]
This is meant to replace https://github.com/wildfly/wildfly/pull/12280.

**Comment by ronsigal** [comment]
Well, I rebuilt wildfly with this PR, ran the failing tests, and they pass. 

I note that 5 of 6 tests fail by returning 200 instead of 400 or 500. The other one expects 400 and gets 500, even though the log suggests that validation discovered an invalid failure. It would help to see the server log in that case.

@asoldano, you have any thoughts?

**Comment by ronsigal** [comment]
And now, when I run testsuite, testsuite/integration, and testsuite/integration/basic in order, the tests fail.

**Comment by ronsigal** [comment]
I built a clean RESTEasy 3.7.0.Final, unzipped it into wildfly's build/module directory, and now the tests pass.

File: feature-pack/pom.xml [hunk 1]
```diff
@@ -1301,7 +1307,7 @@
 
         <dependency>
             <groupId>org.jboss.resteasy</groupId>
-            <artifactId>resteasy-validator-provider-11</artifactId>
```
Comment (marekkopecky) [inline feature-pack/pom.xml]: resteasy-validator-provider-11:3.7.0 artifact doesn't exist anymore? It looks like a backward compatibility blocker (cc @asoldano @msve)

**Review by marekkopecky — CHANGES_REQUESTED** [review]

File: feature-pack/pom.xml (hunk 1 above)
Comment (ronsigal) [reply feature-pack/pom.xml]: Hmmm. I replaced RESTEASY_VALIDATOR_11 in wildfly-jaxrs with RESTEASY_VALIDATOR. Maybe I should have left it and added RESTEASY_VALIDATOR.

Comment (ronsigal) [reply feature-pack/pom.xml]: I restored RESTEASY_VALIDATOR_11.

**Comment by ronsigal** [comment]
The tests went away ...

**Comment by ronsigal** [comment]
Can someone start the tests again?

**Comment by ronsigal** [comment]
Btw, the latest changes have fixed the failures on my laptop.

**Comment by asoldano** [comment]
Retest this please

**Review by asoldano — APPROVED** [review]
The duplicated use of RESTEASY_VALIDATOR and RESTEASY_VALIDATOR_11 is most likely not needed, but it shouldn't hurt, so I'm +1 for merging this as soon as we have green light from CI

**Comment by ronsigal** [comment]
Thanks, @asoldano. I wasn't sure, but I didn't want to wait. I figured out how to get the tests restarted, but they don't make it easy.

File: feature-pack/pom.xml (hunk 1 above)
Comment (bstansberry) [reply feature-pack/pom.xml]: +1. If you'd just left RESTEASY_VALIDATOR_11 my guess is Galleon would see it's an alias module and would provision the target. But my guess is only having RESTEASY_VALIDATOR, Galleon would not know the alias module exists and wouldn't provision it. @jfdenise am I right?

Either way having both constants in there is the most robust.

Comment (jfdenise) [reply feature-pack/pom.xml]: @bstansberry , galleon needs both, there is no support for alias. That is something that would need to be handled by galleon. Will investigate this case.

**Review by bstansberry — APPROVED** [review]
Re: the two RESTEASY_VALIDATOR_11 and RESTEASY_VALIDATOR constants, per the discussion above, both are needed.

**Pull request merged**